src/services/analytics/TagManager.ts

```diff
@@ -38,40 +34,58 @@ const TagManager = {
 
     return script
   },
-  isInitialized: () => {
-    const GTM_SCRIPT = 'https://www.googletagmanager.com/gtm.js'
 
-    return !!document.querySelector(`[src^="${GTM_SCRIPT}"]`)
+  dataLayer: (data: Record<string, any>) => {
+    window[DATA_LAYER_NAME] = window[DATA_LAYER_NAME] || []
+    window[DATA_LAYER_NAME].push(data)
+
+    if (!IS_PRODUCTION) {
+      console.info('[GTM] -', data)
+    }
   },
+
   initialize: (args: TagManagerArgs) => {
-    if (TagManager.isInitialized()) {
-      return
+    window[DATA_LAYER_NAME] = window[DATA_LAYER_NAME] || []
+    // This function MUST be in `window`, otherwise GTM Consent Mode just doesn't work
+    ;(window as any).gtag = function () {
```

Comment:
We can add gtag types: `yarn add -D @types/gtag.js` and add gtag in the `definitions.d.ts` so we don't have to cast here. Also noticed that we have a typo and missing import in the definitions file.